lamindb_setup/core/_hub_core.py

```diff
@@ -82,7 +83,36 @@ def _init_storage(ssettings: StorageSettings, client: Client) -> UUID:
     return id
 
 
-def delete_instance(instance_identifier: str) -> None:
+def delete_instance_by_id(instance_id: UUID) -> None:
+    """Fully delete an instance in the hub.
+
+    This function deletes the relevant instance and storage records in the hub,
+    conditional on the emptiness of the storage location.
+    """
+    from .upath import check_s3_storage_location_empty, create_path, hosted_buckets
+    from ._settings_storage import mark_storage_root
+
+    instance_with_storage = call_with_fallback_auth(
+        select_instance_by_id_with_storage,
+        instance_id=instance_id,
+    )
+
+    if instance_with_storage is not None:
```

Comment:
@bpenteado, you're duplicating a whole section of code here - that's a no-go DRY abuse. Please see 20 lines below and refactor. Or am I missing something?